packages/eslint-plugin-stories/src/rules/no-new-components-without-story.ts

```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+import type { Rule } from 'eslint';
+
+const failureMessage = `
+  React Components must have a corresponding Storybook story.
+
+  "Storybook" is our component explorer, and a "story" is a component example that serves as a unit for Storybook
+
+  Having a Storybook story for each component allows us to test our components' behavior, accessibility,
+  visual regression, etc. Learn more here:
+  https://go.czi.team/storybook-testing
+`;
+
+/**
+ * To encourage Storybook coverage, we warn developers about React components without stories.
+ */
+const rule: Rule.RuleModule = {
+  create(context) {
+    return {
+      ExportDefaultDeclaration(node) {
+        const filename = context.getFilename();
+        const isComponent =
+          filename.match('.tsx') &&
+          !filename.match('.stories.') &&
```

Comment:
Could use `utils/isStories` here. Does the exact same thing as this line, so it doesn't really matter. Could be nice to use the shared util, though 🤷